src/tools/cephfs/top/cephfs-top

```diff
@@ -644,116 +591,209 @@ class FSTop(object):
-            help_text = " [Press 'q' to go back to home (all filesystem info) screen"\
-                        " and 'm' to select another filesystem]"
-
-            # coordinate constants for windowing -- (height, width, y, x)
-            # NOTE: requires initscr() call before accessing COLS, LINES.
-            try:
-                HEADER_WINDOW_COORD = (4, self.width - 1, 0, 0)
-                self.header = curses.newwin(*HEADER_WINDOW_COORD)
-                if fs not in fs_list:
-                    self.stdscr.refresh()
-                    headline = "INFO: Selected filesystem is not available now" + help_text
-                    self.refresh_header(stats_json, headline)
-                else:
-                    if self.height >= 5:
-                        TOPLINE_WINDOW_COORD = (3, self.width - 1, 4, 0)
-                        self.topl = curses.newwin(*TOPLINE_WINDOW_COORD)
-                    else:
-                        self.topl = None
-                    if self.height >= 8:
-                        MAIN_WINDOW_COORD = (self.height - 7, self.width - 1, 7, 0)
-                        self.mainw = curses.newwin(*MAIN_WINDOW_COORD)
-                    else:
-                        self.mainw = None
-                    headline = "HELP: Selected filesystem info" + help_text
-                    if self.refresh_header(stats_json, headline):
-                        x_coord_map = self.refresh_top_line_and_build_coord()
-                        self.refresh_main_window(x_coord_map, stats_json, fs)
-                curses.halfdelay(self.refresh_interval_secs * 10)
-                c = self.header.getch()
-                if c in [ord('m'), ord('q')]:
-                    self.set_option(c)
-            except curses.error:
-                # this may happen when addstr the terminal window
-                # size changed, just retry it
-                pass
+            if fs not in fs_list:
+                help = "Error: The selected filesystem is not available now. " + help_commands
+                self.header.erase()  # erase previous text
+                self.refresh_header(stats_json, help, screen_title, 3)
+            else:
+                self.tablehead_y = 0
+                help = "HELP: " + help_commands
+                self.fsstats.erase()  # erase previous text
+
+                if self.refresh_header(stats_json, help, screen_title, 3):
+                    x_coord_map = self.refresh_top_line_and_build_coord(self.tablehead_y)
+                    self.refresh_clients(x_coord_map, stats_json, fs)
+
+            # scroll and refresh
+            if cmd == curses.KEY_DOWN and \
+               vscrollOffset < self.PAD_HEIGHT - self.viewportHeight - 1:
+                vscrollOffset += 1
+            elif cmd == curses.KEY_UP and vscrollOffset > 0:
+                vscrollOffset -= 1
+            elif (cmd == curses.KEY_RIGHT
+                  and hscrollOffset < self.PAD_WIDTH - self.viewportWidth - 1):
```

Comment:
I can scroll `UP/DOWN` using `KEYS` as well as using two fingers on touchpad, but couldn't scroll `left/right` using two fingers.I am not sure if this can be implemented or not, but we used to have both functioning wherever scroll thing works.It will good to have it also.Just an opinion.

Reply:
I haven't considered enabling the horizontal mouse scroll yet, as there is not much need for it. cephfs-top has fixed number (20) of columns in the table, which doesn't seem difficult to scroll using the arrow keys.  An appropriate terminal font size would fit the entire table too. The vertical scrolling has much more importance because of the greater/unknown number of items to accommodate and scroll.

Reply:
Yeah, it's not much needed.We can ignore.